internal/server/server.go

```diff
@@ -147,3 +160,83 @@ func (s *server) CheckAccess(ctx context.Context, req *authorization.CheckAccess
 		return nil, status.Errorf(codes.Unavailable, err.Error())
 	}
 }
+
+func buildAuthRelations(rels []*authorization.Relationship) ([]events.AuthRelationshipRelation, error) {
+	out := make([]events.AuthRelationshipRelation, len(rels))
+
+	for i, rel := range rels {
+
+		subjID, err := gidx.Parse(rel.SubjectId)
+		if err != nil {
+			return nil, err
+		}
+
+		out[i] = events.AuthRelationshipRelation{
+			Relation:  rel.Relation,
+			SubjectID: subjID,
+		}
+	}
+
+	return out, nil
+}
+
+func (s *server) publishRelationships(ctx context.Context, action events.AuthRelationshipAction, resourceIDStr string, relationships []*authorization.Relationship) error {
+	resourceID, err := gidx.Parse(resourceIDStr)
+	if err != nil {
+		return err
+	}
+
+	relations, err := buildAuthRelations(relationships)
+	if err != nil {
+		return err
+	}
+
+	authReq := events.AuthRelationshipRequest{
+		Action:    action,
+		ObjectID:  resourceID,
+		Relations: relations,
+	}
+
+	s.logger.Infow("request", "req", authReq)
+
+	authResp, err := s.publisher.PublishAuthRelationshipRequest(ctx, authReq)
+	if err != nil {
+		return err
+	}
+
+	if authResp.Error() != nil {
+		return err
+	}
```

Comment:
This should be capturing the error, as the `err` value is already checked on line 203.
```suggestion
	if err := authResp.Error(); err != nil {
		return err
	}
```

Reply:
Good catch!